src/hotspot/share/cds/archiveHeapLoader.cpp

```diff
@@ -136,9 +158,38 @@ class PatchUncompressedEmbeddedPointers: public BitMapClosure {
   }
 };
 
+void ArchiveHeapLoader::patch_compressed_pointers(BitMapView bm,
```

Comment:
The change looks good overall, but I think `patch_compressed_pointers` should be renamed `patch_compressed_embedded_pointers` so that it's clear that this function is a subordinate of `patch_embedded_pointers`.